source/slang/slang-mangled-lexer.cpp

```diff
@@ -0,0 +1,184 @@
+// slang-mangled-lexer.cpp
```

Comment:
Manling and unmangling probably both belong in `mangle.cpp`, or in a paired `name-mangle.cpp` and `name-unmangle.cpp` or similar.